File: frontend/src/types/goal.ts
export enum GoalStatus {
  ON_TRACK = 'ON_TRACK',
  AT_RISK = 'AT_RISK',
  OFF_TRACK = 'OFF_TRACK',
  COMPLETED = 'COMPLETED',
}

export enum GoalMetricType {
  MANUAL = 'MANUAL',
  CARD_COUNT = 'CARD_COUNT',
  STORY_POINTS = 'STORY_POINTS',
}

export interface Goal {
  id: number;
  boardId: number;
  title: string;
  description?: string;
  status: GoalStatus;
  progress: number;
  metricType: GoalMetricType;
}

export interface GoalCreateRequest {
  boardId: number;
  title: string;
  description?: string;
  metricType: GoalMetricType;
}

File: backend/src/main/java/com/kanban/goal/Goal.java
package com.kanban.goal;

import java.time.LocalDateTime;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.kanban.board.Board;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "goals")
@Getter
@Setter
@NoArgsConstructor
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Goal {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "board_id", nullable = false)
    @JsonIgnore
    private Board board;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private GoalStatus status = GoalStatus.ON_TRACK;

    private Integer progress = 0; // 0-100

    @Enumerated(EnumType.STRING)
    @Column(name = "metric_type", nullable = false)
    private GoalMetricType metricType = GoalMetricType.MANUAL;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
